tests/test_grouping.py

```diff
@@ -0,0 +1,341 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the :mod:`grouping` measurement optimization and utility functions.
+"""
+import pytest
+import numpy as np
+from pennylane import Identity, PauliX, PauliY, PauliZ, Hadamard
+from pennylane.operation import Tensor
+from pennylane.wires import Wires
+from pennylane.grouping.group_observables import PauliGroupingStrategy, group_observables
+from pennylane.grouping.graph_colouring import largest_first, recursive_largest_first
+from pennylane.grouping.optimize_measurements import optimize_measurements
+from pennylane.grouping.utils import (
+    is_pauli_word,
+    are_identical_pauli_words,
+    binary_symplectic_inner_prod,
+    pauli_to_binary,
+    binary_to_pauli,
+    is_qwc,
+    convert_observables_to_binary_matrix,
+)
+
+
+class TestGroupingUtils:
```

Comment:
One suggestion for organizing tests would be to mimic the source files and separate the tests based on the particular source they are testing. So, for example, this could go in a `test_utils.py`, there could be a `test_transformations.py` file too, etc.